internal/pkg/agent/install/switch.go

```diff
@@ -76,6 +77,12 @@ func SwitchExecutingMode(topPath string, pt *progressbar.ProgressBar, username s
 		pt.Describe(err.Error())
 	}
 
+	err = EnsureServiceRemoved(30*time.Second, 250*time.Millisecond, paths.ServiceName())
```

Comment:
I see this in `SwitchExecutingMode`, shouldn't it also be in the normal install path, especially the force install case where it has to uninstall the existing agent?

Reply:
yep more than a valid point, ty 🙂 This belongs also in the `Uninstall`. I missed completely that install can uninstall an existing agent as well

Reply:
It doesn't necessarily need to go in Uninstall, it only matters there if agent is going to be installed again. To me, handling it in install makes more sense to avoid waiting on both ends of the process. 

We know this is going to be deleted eventually, and we only care about when it is specifically deleted if we are trying to install the service again.

Reply:
I thought of the same thing, but what if a user uninstalls and then calls install right away?! 🙂 I mean this has to be really tight to happen but still. But with that said, I will make it only be part of the install

Reply:
You could put it in uninstall, just make any error there non-fatal. It will make uninstall take longer unnecessarily if agent is being completely removed from a machine, so my preference is to keep it install where it is necessary but it's not a strong preference.

Reply:
it makes sense to be in install and I added it there as well with dc2e259ee7008d5c142bf692945f77f86f15dbe8 🙂

Reply:
I am gonna resolve this one, please feel free to unresolve if you spot anything else 🙂